lib/validators/post_validator.rb

```diff
@@ -141,6 +142,25 @@ def unique_post_validator(post)
     end
   end
 
+  def force_edit_last_validator(post)
+    return if SiteSetting.max_consecutive_replies == 0 || post.id #|| acting_user_is_trusted?(post) || private_message?(post)
+
+    posts = post.topic&.posts.order('post_number desc').where('not hidden AND posts.deleted_at IS NULL')
```

Comment:
I think you can come up with **one** SQL query that returns the count of the last consecutive posts made by the current user instead of retrieving all the posts and iterating over them in Ruby-land.

Reply:
It's definitely do-able, but I have a feeling it's going to get ugly. 😨